src/plugins/saved_search/common/saved_searches_utils.ts

```diff
@@ -6,36 +6,45 @@
  * Side Public License, v 1.
  */
 
+import { SerializedSearchSourceFields } from '@kbn/data-plugin/common';
 import { SavedSearch, SavedSearchAttributes } from '.';
+import { SerializableSavedSearch } from './types';
 
-export const fromSavedSearchAttributes = (
+export const fromSavedSearchAttributes = <
+  Serialized extends boolean = false,
+  ReturnType = Serialized extends true ? SerializableSavedSearch : SavedSearch
+>(
   id: string | undefined,
   attributes: SavedSearchAttributes,
   tags: string[] | undefined,
-  searchSource: SavedSearch['searchSource'],
-  managed: boolean
-): SavedSearch => ({
-  id,
-  searchSource,
-  title: attributes.title,
-  sort: attributes.sort,
-  columns: attributes.columns,
-  description: attributes.description,
-  tags,
-  grid: attributes.grid,
-  hideChart: attributes.hideChart,
-  viewMode: attributes.viewMode,
-  hideAggregatedPreview: attributes.hideAggregatedPreview,
-  rowHeight: attributes.rowHeight,
-  headerRowHeight: attributes.headerRowHeight,
-  isTextBasedQuery: attributes.isTextBasedQuery,
-  usesAdHocDataView: attributes.usesAdHocDataView,
-  timeRestore: attributes.timeRestore,
-  timeRange: attributes.timeRange,
-  refreshInterval: attributes.refreshInterval,
-  rowsPerPage: attributes.rowsPerPage,
-  sampleSize: attributes.sampleSize,
-  breakdownField: attributes.breakdownField,
-  visContext: attributes.visContext,
-  managed,
-});
+  searchSource: SavedSearch['searchSource'] | SerializedSearchSourceFields,
+  managed: boolean,
+  serialized: Serialized = false as Serialized
+) =>
+  ({
+    id,
+    ...(serialized
+      ? { serializedSearchSource: searchSource as SerializedSearchSourceFields }
+      : { searchSource }),
```

Comment:
Is it because of type checks that we introduce a separate property `serializedSearchSource` which essentially has the same value as `searchSource`?

Reply:
I tried to describe this in my PR description: 

> Previously, when initializing the embeddable, fetching the saved search saved object also created and returned an unserializable search source object. However, in the new system, **runtime state most always be serializable (see https://github.com/elastic/kibana/pull/186052)** - therefore, I've had to instead use the serialized search source in my runtime state saved search - therefore, I've had to make changes to toSavedSearch method to [allow for a serializable saved search to be returned](https://github.com/elastic/kibana/pull/180536/files#diff-3baaeaeef5893a5a4db6379a1ed888406a8584cb9d0c7440f273040e4aa28166R160-R169).

Basically, because runtime state **must be serializable** (this is a requirement we are enforcing on the Dashboard side, so that things can get backed up in session storage + it will allow us to do things like "copy paste" panels in the future), I've had to use `serializedSearchSource` as my runtime state rather than the unserializable `searchSource` :) Hopefully that makes sense?

Reply:
Would not we need to transform `searchSource` to make it serializable?

Reply:
Nope - from the dashboard perspective, it only ever deals with the things defined in `SearchEmbeddableRuntimeState`. You can kind of think of it has having "internal" and "external" runtime state - internal runtime state can have anything you want, serializable or otherwise (`searchSource`, in this case) but **external** runtime state (the stuff the Dashboard cares about) must be serializable (`serializedSearchSource`, in this case)